migrate.go

```diff
@@ -775,6 +817,22 @@ func (m *Migrate) runMigrations(ret <-chan interface{}) error {
 				if err := m.databaseDrv.Run(migr.BufferedBody); err != nil {
 					return err
 				}
+
+				// If there is a post execution function for
+				// this migration, run it now.
+				cb, ok := m.opts.postStepCallbacks[migr.Version]
```

Comment:
IIUC if the post step callback fails, then the version remains dirty. This could end up being a problem requiring down migration from the user. I wonder if it'd be better to just run the migration, the callback and a final `SetVersion` in a transaction?

Reply:
Yeah, I agree. But the thing is: The version is also dirty if the SQL based migration fails, as we also error out then.
And there doesn't seem to be the concept of DB transactions in the migration tool, my guess is because not every supported database backend can do transactions...

So not really sure what to do differently here.

Reply:
> IIUC if the post step callback fails, then the version remains dirty.

This appears to be aa quirk related to the way the library works: https://github.com/golang-migrate/migrate/blob/604248c97e5f61b566c50a8a3261020beeb51489/FAQ.md?plain=1#L62-L65. 

In the context of our desired usage we _typically_ catch situations like this via unit tests of the migration itself.

Reply:
After reading a bit more of the codebase, I think it's possible to run the callback function in the same db transaction as the migration: https://github.com/golang-migrate/migrate/blob/604248c97e5f61b566c50a8a3261020beeb51489/database/sqlite/sqlite.go#L201-L204. 

With the way the interfaces work, if we modify those (using something other than a func opt), then we'd need to update every single driver in the codebase. 

Perhaps the slimmest change would be to add the functional opt to the `Run` method in the main `Driver` interface?